kernel/syscall.c

```diff
@@ -1228,7 +1228,7 @@ static const char* _trim_trailing_slashes(
     {
         memcpy(buf, pathname, len + 1);
 
-        for (char* p = buf + len; p != buf && p[-1] == '/'; *--p = '\0')
```

Comment:
This changes the meaning of the function so that it no longer trims training slashes if there is only one.

So _trim_trailing_slashes("/") => "/" and the trailing slash is not trimmed.